sky/jobs/core.py

```diff
@@ -59,8 +59,10 @@ def launch(
     """
     entrypoint = task
     dag_uuid = str(uuid.uuid4().hex[:4])
-
     dag = dag_utils.convert_entrypoint_to_dag(entrypoint)
+    # Always apply the policy again here, even though it might have been applied
+    # in the CLI. This is to ensure that we apply the policy to the final DAG
+    # and get the mutated config.
```

Comment:
What if the policy is not idempotent (e.g., say the policy is `resources.cpu /= 2`? Should we make sure we apply the policy only once?

Reply:
It should probably be fine to request people to make sure the policy is idempotent. I just added a note in the docs. Wdyt?